app/controllers/miq_task_controller.rb

```diff
@@ -120,8 +120,10 @@ def delete_tasks
   def delete_all_tasks
     assert_privileges("miq_task_deleteall")
     task_ids = []
-    session[:view].table.data.each do |rec|
-      task_ids.push(rec["id"])
+    if session[:view].table.empty? == false
```

Comment:
I'm not sure how your fix could work because table would still be nil, and nil.empty? blows up.

Perhaps instead:

```suggestion
    if session[:view].table.presence?
```